src/main/java/seedu/address/model/good/Good.java

```diff
@@ -22,17 +25,30 @@
     private final GoodQuantity goodQuantity;
     private final GoodQuantity threshold;
 
-    public Good(GoodName goodName, GoodQuantity goodQuantity, GoodQuantity threshold) {
+
+    private final Price transactionPrice;
+
+    public Good(GoodName goodName, GoodQuantity goodQuantity, GoodQuantity threshold, Price transactionPrice) {
```

Comment:
Is this transactionPrice only use for Transaction? i think i have set up sellTransaction that contain a selling price.
Maybe can change the SellCommand constructor to (Good soldGood, Price transactionPrice) instead.
or (SellTransaction sellTransaction) as the sellTransaction have variables: good and sellPrice)
If there is any addtional variable in the Good class, please change the JsonAdaptedGood class accordingly.

Reply:
I was under the impression that sellTransaction was for persistent storage in the transaction history.

There are additional variables in the Good class, but non for persistent storage, only the encapsulate the command args to execute(). This worry might be gone since it might be better to use sellTransaction to encapsulate that info instead.